internal/store/cache_stores_snapshot.go

```diff
@@ -21,6 +27,68 @@ func (c CacheStores) TakeSnapshot() (CacheStores, error) {
 	return snapshot, err
 }
 
+// TakeSnapshotIfChanged takes a snapshot of the CacheStores if the hash of the current state
+// differs from the hash of the previous snapshot supplied as an argument (to make and initial
+// just pass empty string). When error is not nil discard all other return values.
+// When newHash is empty it means that the snapshot hasn't been taken - returned snapshot is
+// meaningless. This is a situation when hash of the current state is the same as the hash of
+// the previous snapshot supplied as an argument.
+func (c CacheStores) TakeSnapshotIfChanged(previousSnapshotHash string) (
```

Comment:
We could consider defining a strong type for the snapshot hash - it would allow us to document its format and the way its calculated + it's a good practice to not pass raw strings.